pkgs/applications/science/electronics/xyce/default.nix

```
@@ -0,0 +1,159 @@
+, withMPI ? false
+  # for doc
+, texlive
+, pandoc
+, enableDocs ? true
+  # for tests
+, bash
+, bc
+, openssh # required by MPI
+, perl
+, perlPackages
+, python38
+, enableTests ? true
+}:
+
+stdenv.mkDerivation rec {
+
+  pname = "xyce";
+  version = "7.4.0";
+  twoLevelVersion = with builtins;
+    concatStringsSep  "." (lib.lists.take 2 (splitVersion version));
+
+  srcs = [
+    # useing fetchurl or fetchFromGitHub doesn't include the manuals
+    # due to .gitattributes files
+    (fetchgit {
+      url = "https://github.com/Xyce/Xyce.git";
+      rev = "82f96bbe05bac5921cd6fa1e8bb6a2983a797bf8";
+      sha256 = "sha256-sOHjQEo4FqlDseTtxFVdLa0SI/VAf2OkwQV7QSL7SNM=";
+    })
+    (fetchurl {
+      url = "https://github.com/Xyce/Xyce_Regression/archive/refs/tags/Release-${version}.tar.gz";
+      sha256 = "00wy2kyvbsyn6gfy42lrkzvgfq9k89hli3ryckywj6p1kcwnvdn9";
+    })
+  ];
+
+  sourceRoot = with builtins; "./Xyce-${substring 0 7 (head srcs).rev}";
+
+  preConfigure = "./bootstrap";
+
+  configureFlags = [ "CXXFLAGS=-O3"
+                     "--enable-xyce-shareable"
+                     "--enable-shared"
+                     "--enable-stokhos"
+                     "--enable-amesos2"
+                   ] ++ lib.optionals withMPI [
+                     "--enable-mpi"
+                     "CXX=mpicxx"
+                     "CC=mpicc"
+                     "F77=mpif77"
+                   ];
+
+  nativeBuildInputs = [ autoconf automake gfortran libtool_2
+                      ] ++ lib.optionals enableDocs [
+                        (texlive.combine {
+                          inherit (texlive)
+                            scheme-medium
+                            koma-script
+                            optional
+                            framed
+                            enumitem
+                            multirow
+                            preprint; })
+                      ];
+
+  buildInputs = [ bison blas flex fftw lapack suitesparse trilinos
+                ] ++ lib.optionals withMPI [ mpi ];
+
+  doCheck = enableTests;
+
+  postPatch = ''
+    pushd ../Xyce_Regression-Release-${version}
+    find Netlists -type f -regex ".*\.sh\|.*\.pl" -exec chmod ugo+x {} \;
+    patchShebangs Netlists/
+    patchShebangs TestScripts/
+    # patch script generating functions
+    sed -i -E 's/\/usr\/bin\/env perl/${builtins.replaceStrings [ "." "/" "-" "!" ] [ "\\." "\\/" "\\-" "\\!" ] perl.outPath}\/bin\/perl/' TestScripts/XyceRegression/Testing/Netlists/RunOptions/runOptions.cir.sh
+    sed -i -E 's/\/bin\/sh/${builtins.replaceStrings [ "." "/" "-" "!" ] [ "\\." "\\/" "\\-" "\\!" ] bash.outPath}\/bin\/sh/' TestScripts/XyceRegression/Testing/Netlists/RunOptions/runOptions.cir.sh
+    popd
+  '';
+
+  checkInputs = [ bc
+                  perl
+                  perlPackages.DigestMD5
+                  (python38.withPackages (ps: with ps; [ numpy scipy ]))
+                ] ++ lib.optionals withMPI [ mpi openssh ];
```

```suggestion
  checkInputs = [
    bc
    perl
    perlPackages.DigestMD5
    (python38.withPackages (ps: with ps; [ numpy scipy ]))
  ] ++ lib.optionals withMPI [ mpi openssh ];
```

Why can't we use python3 which points to python39? python38 is no longer good maintained might just fall apart next week.

---

That's an oversight of mine. Will change that.